rustls/src/quic.rs

```diff
@@ -84,18 +117,41 @@ impl DirectionalKeys {
         };
 
         Self {
-            header: hkdf_expand(secret, hp_alg, b"quic hp", &[]),
+            header: HeaderProtectionKey(hkdf_expand(secret, hp_alg, b"quic hp", &[])),
             packet: PacketKey::new(suite, secret),
         }
     }
 }
 
+/// A key for generating QUIC header protection masks
+pub struct HeaderProtectionKey(aead::quic::HeaderProtectionKey);
+
+impl HeaderProtectionKey {
```

Comment:
Looking at this again with fresh eyes, I wonder why we expose `new_mask` (and `sample_len`)? Why not just have the new encrypt/decrypt functions do the masking too?

I'm guessing we have to separate this because the QUIC implementation needs to unmask in order to keep track of packets, and then do decryption separately?

But, then, still I wonder if this is the right API. Maybe we should have `mask` and `unmask` functions that do the masking and slicing and dicing for a record?

In particular, the return value of `new_mask` is (somewhat) a secret and it would be nice to manage all the secret data within Rustls.

Reply:
I think we could do something like this:

```rust
fn xor(&self, sample: &[u8], first: &mut u8, length: &mut [u8]) -> Result<(), Error>
```

I don't love the name, but can't come up with something better for something that does both masking and unmasking.

Reply:
Went ahead and added this, renaming it to `xor_in_place()`.